File: signature/microsoft_iis.go
package signature

var MicrosoftIisSignature = Signature{
	Name:        "Microsoft IIS",
	Description: "Microsoft Internet Information Services (IIS) is a web server software created by Microsoft for use with the Windows NT family.",
	Cpe:         "cpe:/a:microsoft:iis",

	DetectPattern: DetectPattern{
		Headers: []Header{{
			Name:  "server",
			Value: "Microsoft-IIS",
		}},
	},
	VersionPattern: VersionPattern{
		Headers: []Header{{
			Name:  "server",
			Value: `Microsoft-IIS/(\d+\.\d+)`,
		}},
	},

	OnlySameHost: true,
}

File: src/signatures/technologies/microsoft_iis.ts
import type { Signature } from "../_types.js";

export const microsoftIisSignature: Signature = {
  name: "Microsoft IIS",
  description:
    "Internet Information Services (IIS) is an extensible web server software created by Microsoft for use with the Windows NT family.",
  cpe: "cpe:/a:microsoft:internet_information_server",
  rule: {
    confidence: "high",
    headers: {
      server: "(?:microsoft-)?iis/?(\\d+\\.\\d+)?",
    },
  },
};
